### PR TITLE
Sync Composition v1beta1 with v1

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

We need to keep these two types identical, and unfortunately that process is currently manual.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

No testing for this PR. I did do a quick check for other out-of-sync types by running a few `diff -u v1/foo.go v1beta/foo.go` commands though.

### CODE REVIEW

File: apis/apiextensions/v1beta1/composition_types.go [hunk 1]
```diff
@@ -206,6 +208,7 @@ type PatchPolicy struct {
 	// +kubebuilder:validation:Enum=Optional;Required
 	// +optional
 	FromFieldPath *FromFieldPathPolicy `json:"fromFieldPath,omitempty"`
+	MergeOptions  *xpv1.MergeOptions   `json:"mergeOptions,omitempty"`
```
Comment (hasheddan) [inline apis/apiextensions/v1beta1/composition_types.go:211]: @negz have we had a release with this missing?

**Review by hasheddan — APPROVED** [review]
Thank you for catching this @negz! 👀

File: apis/apiextensions/v1beta1/composition_types.go (hunk 1 above)
Comment (negz) [reply apis/apiextensions/v1beta1/composition_types.go:211]: No, this was only merged within the last day or two.

Comment (hasheddan) [reply apis/apiextensions/v1beta1/composition_types.go:211]: Nice -- thanks!

**Comment by negz** [comment]
Related: https://github.com/crossplane/crossplane/issues/2499

**Pull request merged**